audit_engine/dynamic_analysis/config.py

```
@@ -0,0 +1,161 @@
+    def get_tool_config(self, tool_name: str) -> Dict[str, Any]:
+        """
+        Get configuration for a specific tool with fallback to defaults
+        """
+        tool_configs = {
+            "echidna": self.echidna,
+            "adversarial_fuzz": self.adversarial_fuzz,
+        }
+        
+        return tool_configs.get(tool_name.lower(), {})
+    
+    @lru_cache(maxsize=None)
```

Ruff (0.12.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)